src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -234,22 +234,23 @@ public async Task<bool> TryGetAsync(ArrayBuilder<DiagnosticData> list, Cancellat
                                 : _diagnosticKind == DiagnosticKind.AnalyzerSemantic;
                         }
 
+                        var lazyActiveFileState = new Lazy<ActiveFileState>(() => stateSet.GetOrCreateActiveFileState(_document.Id));
```

Comment:
GetDiagnosticsForSpan: This reduces number of calls from 4 to 1 for non-LSP mode, and from 6 to 1 for LSP mode.